log/zapcore_handler.go

```diff
@@ -0,0 +1,129 @@
+package log
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/prometheus/common/log"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"istio.io/pkg/errdict"
+)
+
+var (
+	toLevel = map[zapcore.Level]Level{
+		zapcore.FatalLevel: FatalLevel,
+		zapcore.ErrorLevel: ErrorLevel,
+		zapcore.WarnLevel:  WarnLevel,
+		zapcore.InfoLevel:  InfoLevel,
+		zapcore.DebugLevel: DebugLevel,
+	}
+	toZapLevel = map[Level]zapcore.Level{
+		FatalLevel: zapcore.FatalLevel,
+		ErrorLevel: zapcore.ErrorLevel,
+		WarnLevel:  zapcore.WarnLevel,
+		InfoLevel:  zapcore.InfoLevel,
+		DebugLevel: zapcore.DebugLevel,
+	}
+)
+
+func init() {
+	RegisterDefaultHandler(ZapLogHandlerCallbackFunc)
+}
+
+// ZapLogHandlerCallbackFunc is the handler function that emulates the previous Istio logging output and adds
+// support for errdict package and structured logging.
+func ZapLogHandlerCallbackFunc(
+	level Level,
+	scope *Scope,
+	ie *errdict.IstioErrorStruct,
+	msg string,
+	fields []zapcore.Field) {
+	var ss []string
+	if ie != nil {
+		ss = append(ss, fmt.Sprintf(`"moreInfo":"%s"`, ie.MoreInfo))
+		ss = append(ss, fmt.Sprintf(`"impact":"%s"`, ie.Impact))
```

Comment:
hand crafting json with arbitrary inputs means the output may easily become invalid json

Reply:
i think we can do better here to avoid that. i'd rather not include a json library to format the json though. 
how about, number types have no quotes, everything else has escaped quotes? then objects will not cause invalid json, only messy json :-)